lib/dice/src/handoff.rs

```diff
@@ -38,26 +56,68 @@ impl<'a> Handoff<'a> {
         self.0.ahbclkctrl2.modify(|_, w| w.usb1_ram().disable());
     }
 
-    pub fn store(&self, alias_data: &AliasData) -> usize {
-        // SAFETY: Dereferencing this raw pointer is necessary to write to the
-        // memory region used to handoff DICE artifacts to Hubris tasks. This
-        // pointer will references a valid memory region provided two
-        // conditions are met:
-        // 1) The associated memory region has been enabled / turned on if
-        // necessary. This happens in the constructor / 'turn_on' function.
-        // 2) The function call is made by code sufficintly privileged to
-        // access the memory region (e.g. stage0).
-        // If these conditions aren't met this access is still safe but a fault
-        // will occur.
-        let dst: &mut [u8] = unsafe {
+    pub fn store<T>(&self, t: &T) -> usize
+    where
+        T: HandoffData + SerializedSize + Serialize,
+    {
+        // Cast MEM_RANGE from HandoffData to a mutable slice.
+        //
+        // SAFETY: This unsafe block relies on implementers of the HandoffData
+        // trait to validate the memory range denoted by Self::MEM_RANGE. Each
+        // implementation in this module is checked by static assertion.
+        let dst = unsafe {
             core::slice::from_raw_parts_mut(
-                ALIAS_START as *mut u8,
-                AliasData::MAX_SIZE,
+                T::MEM_RANGE.start as *mut u8,
+                <T as SerializedSize>::MAX_SIZE,
             )
         };
-
         // TODO: error handling
-        hubpack::serialize(dst, alias_data).expect("serialize alias-handoff")
+        hubpack::serialize(dst, t).expect("handoff store")
+    }
+}
+
+// Types that can be transfered through the memory region used to pass DICE
+// artifacts from stage0 to hubris tasks.
+//
+// This trait cannot check the validity of the memory range selected by
+// implementers and so implementers of this trait are required to ensure that
+// the range denoted by Self::MEM_RANGE is:
+// - within the memory range used to hold DICE artifacts
+// - non-overlapping with the ranges of memory used by other implementers of
+// this trait
```

Comment:
- large enough to contain a the largest serialized form of the implementing type

You already are checking this in your static asserts but didn't mention it here.